pkg/k8shandler/fluentd.go

```diff
@@ -414,6 +414,37 @@ func newFluentdPodSpec(cluster *logging.ClusterLogging, elasticsearchAppName str
 	// Shorten the termination grace period from the default 30 sec to 10 sec.
 	fluentdPodSpec.TerminationGracePeriodSeconds = utils.GetInt64(10)
 
+	// FIXME: The following conditional branch is a refactoring candidate
+	//
+	// Why do we need this branch at all?
+	//
+	// The ClusterLogging CR supports three additional use cases without providing
+	// a logStore stanza namely:
+	// Case 1: A collection only stanza to use with secure forward
```

Comment:
Why do we specifically need to call out the target (e.g fluent or secure forward)?  It is only needed for the default output we provide; this is the only case

Reply:
Describing only the positive case where a log store is used with/without LF in place leaves a lot of space to forget why we support a collection-only stanza in ClusterLogging. According to docs we support only three use cases. In addition calling them out here puts more pressure to the reader to refactor this code branch, don't you think?